### DISCUSSION TITLE
Is it possible to configure build config like javaOptions from worksheet.sc?

### DISCUSSION THREAD

**Discussion by i10416** [scalameta/metals#3345] [Q&A] [answered]
Hello, I wonder whether it is possible to configure build properties from worksheet.sc file.

What I want to do is to set properties(e.g. `Dfoo.bar=baz`)  or java options (e.g. `javaOptions+=-Djna.library.path=...`) from worksheet file instead of build.sbt.

foo.worksheet.sc

```scala
// -Dfoo.bar=baz
// javaOptions+=-Djna.library.path=...

import com.example.lib
val x = lib.foo // This depends on properties or javaOptions for initialization.
```

AFAIK, we can set Scala version and ammonite version in header comment.

```scala
// scala 2.13.7
// ammonite 2.4.x
```

However, I couldn't find the way to set properties or java options.  Is it possible to configure build config like javaOptions from worksheet.sc?

**Comment by tgodzik** [comment, marked as answer]
It's not possible currently to set java options (only scalac ones are possible) as the worksheet exists within Metals JVM. You should be able to modify `metals.serverProperties` and that should also set it for the worksheet.

**Reply by i10416** [reply]
Thank you!
